### DISCUSSION TITLE
[Bug]: ImageMagick - UnableToOpenConfigureFile When Attempting To Read SVG File

### DISCUSSION THREAD

**Discussion by T-Spoon** [beyondcode/herd-community#90] [Q&A]
### Operating system version

macOS Monterey 12.5.1 (21G83)

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.0.8 (Build: 10)

### PHP Version

PHP 8.2.8 (cli) (built: Jul 28 2023 13:55:05) (NTS)

### Bug description

Attempting to use ImageMagick with SVG support and the following error is encountered:

```
UnableToOpenConfigureFile `delegates.xml' @ warning/configure.c/GetConfigureOptions/722
````

The locally installed version of ImageMagick is `7.1.1-10`, and I've also added an alias for `7.1.1-14` to point to my `7.1.1-10` install, as I can see that the bundled version with herd is `7.1.1-14`.  

Is there some way of configuring Herd's ImageMagick params? Since AFAIU all the extensions are compiled in the Herd php binary - I'm not sure how to go about debugging this issue.

### Steps to reproduce

```
$image = new Imagick();
$svg = file_get_contents("image.svg");
$image->readImageBlob($svg);
```

### Relevant log output

```shell
UnableToOpenConfigureFile `delegates.xml' @ warning/configure.c/GetConfigureOptions/722
```

**Comment by MACscr** [comment]
ddi you have any luck getting this resolved? I got the same error in the latest herd as well

**Comment by MACscr** [comment]
On top of this, I need to add HEIC and AVIF support as well. Excited to Herd to mature to make these type of things easier.

**Comment by Kazuto** [comment]
I wanted to use Herd as well but I can't because the project I work in needs this ImageMagick feature. Have yet to find a solution for it. In the meantime I'm stuck with Valet.

**Comment by ahartmann-preeco** [comment]
Same issue here. Any ideas on how to solve this issue?

**Comment by marcorieser** [comment]
Why is this in discussions?

**Comment by robdekort** [comment]
Hey folks. I'm also running into this issue. I'm noticing it on both PHP 8.1, 8.2 and 8.3. This seems like an issue with Herd so I also wonder why the issue got converted to a discussion. Nice job with Herd, but wouldn't it be more helpful for folks to see at a glance which limitations there still are with the software?

If someone knows a workaround I'd be happy to learn.

Best!

**Reply by crazywhalecc** [reply]
The problem may be that Herd's static php builder has not enabled `--with-xml` for the `imagemagick` library. I have fixed this problem within the past two months, but the official Herd builder may not have applied this update.

**Reply by robdekort** [reply]
Thanks for your comment. Much appreciated. I see. So if they were to update the builder and recompile the binaries to include the xml-files, the delegates.xml file would be present.

I've been trying to install imagick with pecl manually and linking to it from the herd php.ini file, but it seems I can't overwrite extensions.
